units/patch_intent.go

```diff
@@ -306,6 +306,10 @@ func (j *patchIntentProcessor) finishPatch(ctx context.Context, patchDoc *patch.
 		}
 	} else {
 		var patchConfig *model.PatchConfig
+		repeatPatchID, shouldRepeat := j.intent.RepeatPreviousPatchDefinition()
+		if shouldRepeat {
+			patchDoc.RepeatPatchID = repeatPatchID
```

Comment:
Should we move the logic above that gets the patch and resolves the module's git hashes to here? I think it would fail earlier maybe but I didn't read the logic fully. It would also eliminate the RepeatPatchID on the patch document, which I think would reduce clutter.

Reply:
I'd argue against it because of the complexity of this function already and it's doing a lot 

I also think it makes sense for it to be in the getModule function